### PR TITLE
Updates to fonts and colors styles

### PR DESCRIPTION
**Pull request by dephillipsmichael**
There were a few missing font and color stylings on the active step vc, and it seemed appropriate to create a font type for buttons so I created that and applied it to the RSDRoundedButton.

### CODE REVIEW

File: Research/Research/RSDFontRules.swift [hunk 1]
```diff
@@ -102,6 +102,19 @@ open class RSDFontRules  {
...
+    /// - parameter buttonType: The button type for the font.
+    /// - returns: The font to use for this button text.
+    open func font(for buttonType: RSDDesignSystem.ButtonType) -> RSDFont {
+        switch buttonType {
+        case .primary:
+            return RSDFont.systemFont(ofSize: 20, weight: .bold)
+        case .secondary:
+            return RSDFont.systemFont(ofSize: 18, weight: .bold)
+        }
+    }
```
Comment (syoung-smallwisdom) [inline Research/Research/RSDFontRules.swift:117]: I will confirm with Stockard tomorrow that this is the desired behavior.

Comment (dephillipsmichael) [reply Research/Research/RSDFontRules.swift:117]: See https://www.figma.com/file/nvoSigSxbFuWzGXgUZAf8M/DigitalHealth_DesignSystem-Master?node-id=3125%3A322

**Comment by dephillipsmichael** [comment]
@syoung-smallwisdom also I had a conversation with Woody and the cancel button image is now going to be tinted the color of the secondary button text, and not the primary button background color.  What do you think is the best way to go about making this change?

**Comment by syoung-smallwisdom** [comment]
@dephillipsmichael 
As for changing the color rules, rev the version number and change the rule, checking for version number to keep the previous rule in play for previous versions.

WRT meeting with Stockard, that was moved to this Wednesday. Will be confirming changes to font and color rules before accepting changes. Sorry for the delay.

File: Research/Research/RSDFontRules.swift (hunk 1 above)
Comment (syoung-smallwisdom) [reply Research/Research/RSDFontRules.swift:117]: So... Stockard and I discussed how fonts for the rounded buttons should *not* be dynamic b/c that messes with the sizing of the button, and that the text on the button can be made accessible by turning on voice-over instead.

Toward that end, we talked about introducing a method that is more generic. Namely, doing this:

```
    /// Returns a font in the given size and weight in the font family specified for this design. Typically, you will
    /// want to use `font(for textType: RSDDesignSystem.TextType)` instead for copy and dynamic text. This method should
    /// only be used where the design calls for a specific size to match the graphic design of the view.
    ///
    /// - parameters:
    ///     - fontSize: The font size.
    ///     - weight: The font weight.
    /// - returns: The font to use for this size and weight.
    open func font(ofSize fontSize: CGFloat, weight: RSDFont.Weight) -> RSDFont {
        return RSDFont.systemFont(ofSize: fontSize, weight: weight)
    }
```

And then use *this* method for elements that should be a statically defined size/weight (such as the rounded button).

The frameworks/applications can then set the font family or whatever to override the base class (although eventually we want to include Lato in this bundle.) 

Also, this leaves room in the designs to eventually accommodate vision impaired folks who prefer larger text.

File: Research/ResearchUI/iOS/Step View Controllers/RSDActiveStepViewController.swift [hunk 2]
```diff
@@ -254,4 +254,20 @@ open class RSDActiveStepViewController: RSDFullscreenImageStepViewController {
...
+        
+        super.setColorStyle(for: placement, background: background)
+        
+        if (placement == .body) {
+            self.countdownLabel?.font = self.designSystem.fontRules.font(for: .counter)
+            self.countdownLabel?.textColor = self.designSystem.colorRules.textColor(on: background, for: .counter)
+            
+            self.unitLabel?.font = self.designSystem.fontRules.font(for: .body)
+            self.unitLabel?.textColor = self.designSystem.colorRules.textColor(on: background, for: .body)
+        }
```
Comment (syoung-smallwisdom) [inline Research/ResearchUI/iOS/Step View Controllers/RSDActiveStepViewController.swift:271]: These are currently statically defined b/c as graphical elements, they need to fit within the circle. To allow for customization of the font family, instead they should use the `font(forSize:weight:)` method described above.

File: Research/ResearchUI/iOS/Step View Controllers/RSDCountdownStepViewController.swift [hunk 3]
```diff
@@ -101,6 +101,7 @@ open class RSDCountdownStepViewController: RSDFullscreenImageStepViewController
         super.setColorStyle(for: placement, background: background)
         if placement == .body {
             countdownLabel?.textColor = countdownLabelColor(on: background)
+            self.countdownLabel?.font = self.designSystem.fontRules.font(for: .counter)
```
Comment (syoung-smallwisdom) [inline Research/ResearchUI/iOS/Step View Controllers/RSDCountdownStepViewController.swift:104]: This should be statically defined b/c as a graphical element, it needs to fit the view. To allow for customization of the font family, instead it should use the `font(forSize:weight:)` method described above.

File: Research/ResearchUI/iOS/Views/RSDRoundedButton.swift [hunk 4]
```diff
@@ -155,7 +155,7 @@ import UIKit
         }
         
         // Set the title font to the font for a rounded button.
-        titleLabel?.font = designSystem.fontRules.font(for: .heading4)
+        titleLabel?.font = designSystem.fontRules.font(for: buttonType)
```
Comment (syoung-smallwisdom) [inline Research/ResearchUI/iOS/Views/RSDRoundedButton.swift:158]: This should be statically defined to fit the button. Maybe as a property?

**Review by syoung-smallwisdom — CHANGES_REQUESTED** [review]
After speaking with Stockard, we discussed including a more general method rather than trying to come up with a "named" text type for every text element used.

**Comment by syoung-smallwisdom** [comment]
Closing. This is covered by https://github.com/Sage-Bionetworks/SageResearch/pull/129

**Pull request closed without merging**